noxfile.py

```diff
@@ -136,9 +135,16 @@ def tests_compiler(session: nox.Session, version: str) -> None:
     set_environment(session, version)
     session.log("Print tool versions")
     session.run("python", "--version")
-    session.run(session.env["CC"], "--version", external=True)
-    session.run(session.env["CXX"], "--version", external=True)
-    session.run(*shlex.split(session.env["GCOV"]), "--version", external=True)
+    # Use full path to executable
+    for env in ["CC", "CXX", "GCOV"]:
+        value = shlex.split(session.env[env])
+        value[0] = shutil.which(value[0])
```

Comment:
I'm not quite sure why this would be needed but it doesn't seem incorrect.

Reply:
This is for using always full paths in the test. Know we also have the full paths to the used executables in the test log.

Reply:
Resolving the full paths is fine. But as discussed in my other comment <https://github.com/gcovr/gcovr/pull/541#discussion_r775275821>, I'm not convinced about adding shell escaping to the CC and CXX variables. This seems to complicate stuff throughout the tests, and does not seem to provide significant value. Shell escaping is not necessary for supporting full paths, assuming that `C:/` style paths with forward slashes work on Windows. If someone were to use a path that e.g. contains spaces or shell metacharacters, something would probably break anyway.

That shell escaping is needed for GCOV is solely due to Clang support.